File: wiki-backend/app/schemas/media.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from datetime import datetime
from typing import List, Optional
from enum import Enum

class MediaFileType(str, Enum):
    ALL = "all"
    IMAGE = "image"
    VIDEO = "video"
    AUDIO = "audio"
    PDF = "pdf"
    TEXT = "text"

class MediaBase(BaseModel):
    original_filename: str
    bucket_name: str
    object_key: str
    mime_type: str
    file_size: int

class MediaCreate(MediaBase):
    public_url: Optional[str] = None

class MediaResponse(MediaBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    storage_path: str
    public_url: str
    uploaded_at: datetime

class MediaListResponse(BaseModel):
    data: List[MediaResponse]
    total: int


class MediaInfoResponse(MediaResponse):
    articles: List[dict] = []
    commits: List[dict] = []
    is_orphaned: bool

class MediaUploadResponse(MediaResponse):
    message: str = "File uploaded successfully"

class ArticleReference(BaseModel):
    id: UUID
    title: str

class CommitReference(BaseModel):
    id: UUID
    message: str

File: frontend/src/api/types/media.ts

// types/media.ts
export interface MediaResponse {
  id: string;
  original_filename: string;
  storage_path: string;
  bucket_name: string;
  object_key: string;
  mime_type: string;
  file_size: number;
  public_url: string;
  uploaded_at: string;
}

export interface MediaUsageResponse {
  articles: { id: string; title: string }[];
  commits: { id: string; message: string }[];
  is_orphaned: boolean;
}

// Add the missing types
export interface MediaFile extends MediaResponse {
  // MediaFile is the same as MediaResponse, we can use it interchangeably
  // or add any additional fields if needed
}

export interface MediaUploadResponse {
  id: string;
  original_filename: string;
  storage_path: string;
  bucket_name: string;
  object_key: string;
  mime_type: string;
  file_size: number;
  public_url: string;
  uploaded_at: string;
  message: string;
}

export interface MediaInfoResponse extends MediaResponse {
  articles: { id: string; title: string }[];
  commits: { id: string; message: string }[];
  is_orphaned: boolean;
}

// Types for API responses
export interface MediaListResponse {
  data: MediaFile[];
  total: number;
  page?: number;
  pages?: number;
}

export interface MediaDownloadResponse {
  download_url: string;
  expires_in: number;
}

export const MediaFileType = {
  ALL: 'all',
  IMAGE: 'image',
  VIDEO: 'video',
  AUDIO: 'audio',
  PDF: 'pdf',
  TEXT: 'text'
} as const;

export type MediaFileType = typeof MediaFileType[keyof typeof MediaFileType];

// Update MediaListParams to use the enum
export interface MediaListParams {
  page?: number;
  skip?: number;
  limit?: number;
  search?: string;
  type?: MediaFileType | string; // Allow string for backward compatibility
}


export interface UploadMediaParams {
  file: File;
  articleId?: string;
  commitId?: string;
}

export interface AttachMediaParams {
  mediaId: string;
  articleId: string;
  commitId?: string;
}

export interface DetachMediaParams {
  mediaId: string;
  articleId: string;
}

export interface DownloadMediaParams {
  mediaId: string;
  expiresIn?: number;
}